lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -78,54 +78,62 @@ def self.deliver(*args)
     begin
       miq_task&.state_active
       object_name = "#{options[:object_type]}.#{options[:object_id]}"
-      _log.info("Delivering #{ManageIQ::Password.sanitize_string(options[:attrs].inspect)} for object [#{object_name}] with state [#{state}] to Automate")
       automate_attrs = automate_attrs_from_options(options)
-
       if options[:object_type]
         vmdb_object = options[:object_type].constantize.find_by!(:id => options[:object_id])
         automate_attrs[create_automation_attribute_key(vmdb_object)] = options[:object_id]
         vmdb_object.before_ae_starts(options) if vmdb_object.respond_to?(:before_ae_starts)
         vmdb_object.mark_execution_servers if vmdb_object.respond_to?(:mark_execution_servers)
       end
-
       uri = create_automation_object(options[:instance_name], automate_attrs, create_automation_object_options(options, vmdb_object))
       ws  = resolve_automation_object(uri, user_obj)
+      miq_request_id = nil
+
+      if !defined?(ws.root.attributes).nil?
```

Comment:
not sure if `defined?` is the correct choice here - what is this trying to check?

Reply:
Just wants to make sure `ws.root.attributes` exists so that it doesn't run into problems trying to grab `ws.root.attributes['miq_request_id']` for example

Reply:
Yeah defined is not the right way to do this... Think the following will work